book/src/rfcs/ffadd.md

```diff
@@ -0,0 +1,192 @@
+
+Alternatively, one can store negative values $-e$ as a negative sign followed by the absolute value $e$. Doing so will affect the way we address foreign field additions in the case that we had $a - b$ or $b - a$ (note that $- a - b$ is just the negation of $a + b$, and nothing changes at the gate level). Specifically, the carries between limbs may not just be bits ($0$ or $1$), but they could also be $-1$. This change only affects the values of $c_0, c_1, c_2$. 
+
+For the second part of the gate, meaning the sum for the range check, the carry bits will remain unchanged. The reason behind this is the fact that $g > 0$ because $2^{264} > f$, and $r$ will be "positive" as well. That means, this sum follows the usual positive-addition carry structure. 
+
+## Gadget
+
+The foreign field gadget will be composed by 4 sets of `RangeCheck` gadgets for witnesses $a, b, r, s$ accounting for 16 rows in total; followed by one row with `ForeignFieldAdd` gate type; and a final `Zero` row. A total of 18 rows with 15 columns in Kimchi.
+
+| Row(s) | Gate type(s)        | Witness |
+| ------ | ------------------- | ------- |
+| 0-3    | `multi-range-check` | $a$     |
+| 4-7    | `multi-range-check` | $b$     |
+| 8-11   | `multi-range-check` | $r$     |
+| 12-15  | `multi-range-check` | $s$     |
+| 16     | `ForeignFieldAdd`   |         |
+| 17     | `Zero`              |         |
+
+
+### Layout
+
+For this gate, we need to perform 4 range checks to assert that the limbs of $a, b, r, s$ have a correct size, meaning they fit in $2^{88}$ (and thus, range-checking $a, b, r, s$ for $2^{264}$). Because each of these elements is split into 3 limbs, we will have to use 3 copy constraints between the `RangeCheck` gates and the `ForeignFieldAdd` rows (per element). That amounts to 12 copy constraints. Recalling that Kimchi only allows for the first 7 columns of each row to host a copy constraint, we necessarily have to use 2 rows for the actual addition gate. The layout of these two rows is the following:
+
+|            | Curr              | Next         |
+| ---------- | ----------------- | ------------ |
+| **Column** | `ForeignFieldAdd` | `Zero`       |
+| 0          | $a_0$ (copy)      | $r_0$ (copy) |
+| 1          | $a_1$ (copy)      | $r_1$ (copy) |
+| 2          | $a_2$ (copy)      | $r_2$ (copy) |
+| 3          | $b_0$ (copy)      | $u_0$ (copy) |
+| 4          | $b_1$ (copy)      | $u_1$ (copy) |
+| 5          | $b_2$ (copy)      | $u_2$ (copy) |
+| 6          | $x$               |
+| 7          | $c_0$             |
+| 8          | $c_1$             |
```

Comment:
Don't we need c2?

Reply:
Actually no, because we are adding foreign fields together and this addition should not be larger than 2^264 in any case. Meaning, we could have a `c_2` and constrain it to be always zero. But not having it is implicitly constraining that same thing.